src/System.Net.Security/src/System/Net/SecureProtocols/_SslState.cs

```diff
@@ -0,0 +1,1745 @@
+            }
+        }
+
+        internal CipherAlgorithmType CipherAlgorithm
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return CipherAlgorithmType.None;
+                }
+                return (CipherAlgorithmType)info.DataCipherAlg;
+            }
+        }
+
+        internal int CipherStrength
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return 0;
+                }
+                return info.DataKeySize;
+            }
+        }
+
+        internal HashAlgorithmType HashAlgorithm
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return (HashAlgorithmType)0;
+                }
+                return (HashAlgorithmType)info.DataHashAlg;
+            }
+        }
+
+        internal int HashStrength
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return 0;
+                }
+                return info.DataHashKeySize;
+            }
+        }
+
+        internal ExchangeAlgorithmType KeyExchangeAlgorithm
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return (ExchangeAlgorithmType)0;
+                }
+                return (ExchangeAlgorithmType)info.KeyExchangeAlg;
+            }
+        }
+
+        internal int KeyExchangeStrength
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return 0;
+                }
+                return info.KeyExchKeySize;
+            }
+        }
+
+        internal SslProtocols SslProtocol
+        {
+            get
+            {
+                CheckThrow(true);
+                SslConnectionInfo info = Context.ConnectionInfo;
+                if (info == null)
+                {
+                    return SslProtocols.None;
+                }
+
+                SslProtocols proto = (SslProtocols)info.Protocol;
+
+                // Restore client/server bits so the result maps exaclty on published constants.
+                if ((proto & SslProtocols.Ssl2) != 0)
+                {
+                    proto |= SslProtocols.Ssl2;
+                }
```

Comment:
I don't understand what these ifs are doing.

If the SSL2 bit is set, set the SSL2 bit.  Repeat for further protocols.  The comment sounds like it's trying to clear out unknown bits; but that seems like you want a read-from enum variable and a write-to enum variable...